src/conda_project/utils.py

```diff
@@ -182,3 +183,14 @@ def detect_shell():
             )
 
     return shell_name, shell_path
+
+
+def dedupe_list_of_dicts(data: list, key: Callable, keep: Callable) -> list:
```

Comment:
This may be an optimization we can try later, but we may be able to use a generator to iterate over in the for loop that calls this and avoid instantiating the list again.

Reply:
Good point. I'll make an issue for later.